### DISCUSSION TITLE
monospaced font for macro/shell editor [FIXED]

### DISCUSSION THREAD

**Discussion by tksoh** [eteran/nedit-ng#234] [General]
The editing area of the shell and macro customization dialogs is using variable font currently. I think it should be changed to use the monospaced font, perhaps just follow default font setting. NEdit 5 use monospaced font.

I also wonder if it's possible to port in the main text editor widget, that would make the macros development work a lot more productive. Right now it's basically just a highly primitive text editor (same on nedit 5), which doesn't quite live up to NEdit's reputation ;-)

**Comment by eteran** [comment]
Interesting. On my machine, those dialogs use a monospace font. Specifically, the UI files ask for a font called "Monospace", LOL. So it must be a font availability issue.

I'll first look into what the best way to force a monospace font on windows. But you're right, using the main NEdit editor widget is a good idea, as we'll get syntax highlighting and all the other goodness with it!

BTW, I think this one probably should be an "issue" since it is something that should be fixed :-). Makes it easier to track things as "close" after the work is complete.

**Comment by eteran** [comment]
@tksoh when you get a chance, could you check to see if this PR fixes this problem?

https://github.com/eteran/nedit-ng/pull/235

There were actually a bunch of dialogs that are **intended** to have fixed-width fonts beyond the two you mentioned!

**Comment by tksoh** [comment]
Looks good.

Now, let's add syntax highlight and other goodies to it ;-)

**Comment by tksoh** [comment]
> BTW, I think this one probably should be an "issue" since it is something that should be fixed :-). Makes it easier to track things as "close" after the work is complete.

At times, I could be unsure if an issue is indeed an issue, or a design decision. But we can always 'promote'  a discussion to an issue if you think it should be, and continue our discussion there ;-)

**Reply by eteran** [reply]
So Github has an automatic way to convert an issue to a discussion, but not vice-versa.

So I'd say if you're looking for something to be "fixed", then probably best to consider it an issue. And if it ends up being more a discussion, it can easily be converted to one :-).

**Reply by tksoh** [reply]
sounds good

**Reply by sjtringali** [reply]
This was intentional in 5.x. Text fields all used fixed-pitch fonts.

**Comment by grege2** [comment]
My Mac version is using a proportional font, for all Dialogs.
But I like it fine actually. 
Example below ..
![menuimage](https://user-images.githubusercontent.com/33416604/103391742-0a3aa280-4b6f-11eb-9045-7c2f32a8e5d2.png)

**Reply by eteran** [reply]
@grege2 Is that using the latest master? The PR that was just merged **should** have "fixed" that.

**Comment by grege2** [comment]
Evan, we crossed in the ether. Yes my comment was an earlier master.

I was just building the latest for #235 on Mac, that you announced, and the macro and shell command areas are now monospaced. Definitely more sensible for "code". They seem fixed at about 11 or 12-point Courier, ok but a touch small. Not sure if that can be changed, I can't see any "monospaced" settings in nedit-ng. You say nedit-ng just asks the OS for "Monospace" ?

**Reply by eteran** [reply]
Yea, the font size I think is whatever the system default is. It's definitely possible to make it configurable, just not sure where that should go. I'll have to think on it.

As for your question, yea, in the .ui file, the font-family was set to "Monospace", which on linux systems "just works". Now I've edited it to actually ask the `QFontDatabase` for the system-specific "TypeWriter" font (which is, I guess, font-speak" for a fixed with font suitable for code). That seems to be a pretty good cross-platform way to do it.

I could of course also do like @tksoh suggests and just have it use whatever font the user has chosen for their editor too. That actually makes a lot of sense. I could even use the size they've chosen too!

**Reply by tksoh** [reply]
I think picking up both font and size from default setting is quite sensible. If I am using font X or size M for my regular editing, then I would want to do the same in the macro/shell editor also. After all, that's still programming, but in another language.

**Comment by tksoh** [comment]
I earlier suggested following default font setting. Not sure if that's possible, or there might other complication.

**Comment by eteran** [comment]
@tksoh it's definitely possible. honestly, the next step is to use the same text widget for macros as we do for nedit's main editor. Then we can have all the goodies like you pointed out.

**Reply by tksoh** [reply]
It'd be great If we could get that done soon :-)
